### PR TITLE
refactor: remove unused wait=False path from build_systems

### LINKED ISSUE #52: refactor: remove unused wait=False path from build_systems
## Problem

`build.py` has the same `wait=False` / `session.detach()` pattern that was just removed from `simulate.py` in PR 50:

```python
if not wait:
    logger.warning("Returning raw futures — caller must call parsl.clear() when done.")
    session.detach()
    return futures
```

Same issues apply:
- **No callers use it** — every call site passes `wait=True` (the default)
- **Broken return type** — returns raw `AppFuture` objects that callers can't meaningfully use without managing the Parsl session themselves
- **Leaks session** — `session.detach()` leaves the Parsl session open with no cleanup guarantee

## Proposed fix

- Remove the `wait` parameter from `build_systems`
- Remove the 4-line `if not wait` block and `session.detach()` call
- Remove the `wait` parameter from the docstring
- Update any tests that use `wait=False`

~15 lines removed, no behavioral change for any existing caller.

## Acceptance criteria

- [ ] `wait` parameter removed from `build_systems` signature and docstring
- [ ] `if not wait` block and `session.detach()` call removed
- [ ] All existing tests pass
- [ ] No behavioral changes

### PR DESCRIPTION
**Pull request by gregorweiss**
Closes #52

Remove unused `wait=False` path from `build_systems`, matching the cleanup done in PR 50 for `simulate.py`. Also removes `session.detach()` from `ParslSession` since no caller uses it after this change.

Implementation plan posted as a comment below.

### CODE REVIEW

**Comment by gregorweiss** [comment]
## Implementation Plan

### Problem
`build_systems` has an unused `wait=False` / `session.detach()` path identical to the one removed from `simulate.py` in PR 50. No caller uses it, it returns raw AppFutures, and it leaks the Parsl session.

### Deliverables

#### 1. `mdfactory/orchestration/build.py`
- Remove `wait` parameter from `build_systems` signature
- Remove `wait` from docstring (parameter + returns sections)
- Remove the 4-line `if not wait` block and `session.detach()` call

#### 2. `mdfactory/orchestration/session.py`
- Remove `detached` field from `ParslSession` dataclass
- Remove `detach()` method from `ParslSession`
- Remove the `if not session.detached` guard in `parsl_session` context manager — always run cleanup
- Remove the `wait=False` reference in `detach()` docstring

#### 3. `mdfactory/tests/test_orchestration_build.py`
- Remove `test_build_systems_no_wait` (line 134)
- Remove `test_build_systems_no_wait_does_not_clear` (line 402)

### Verification
- No CLI callers use `wait=False` — all 4 call sites in `cli.py` pass only `dry_run` and `output_dir`
- Run existing build + session test suites
- No new tests needed — only dead code removal

### Risk
Near zero — mirrors exactly what was done in PR 50 for `simulate.py`.

---
*Plan created by mach6*

**Comment by gregorweiss** [comment]
## Progress Update

Removed the unused `wait=False` path from `build_systems` and `detach()` from `ParslSession`:

- **`build.py`**: removed `wait` parameter, `if not wait` block, `session.detach()` call, unused `session` binding
- **`session.py`**: removed `detached` field, `detach()` method, conditional shutdown guard (now always cleans up)
- **`test_orchestration_build.py`**: removed 2 `wait=False` tests
- **`test_orchestration_session.py`**: removed `detach` test and `detached` assertion

4 files changed, 6 insertions, 97 deletions. 20 tests pass, lint clean.

Note: 2 pre-existing test failures in `test_shutdown_parsl_*` are unrelated (squeue safety net calls `subprocess.run` twice but test asserts once).

**Commit:** `fdfc263`

---
*Progress tracked by mach6*

**Comment by gregorweiss** [comment]
## Code Review

### Critical
None.

### Important
None.

### Suggestions

**Finding 1** (simplifier, low): Leftover "Finding N" code-review reference comments in `test_orchestration_build.py` (e.g. `# --- Finding 10: failure description helper ---`). These are review-thread artifacts that carry no meaning — the subject-describing part is useful, the "Finding N" prefix is noise.

**Finding 2** (simplifier, medium): Duplicate `"error"` and `"error_detail"` keys holding identical values in `_poll_future` result dict (`build.py` line 213). Both are always set to the same string.

**Finding 3** (simplifier, low): Module-level imports placed mid-file in `build.py` with `# noqa: E402` suppressions (`from .errors import _describe_failure` and `from .progress import ...`). Could be lifted to the top-level import block.

### Strengths
- Clean, surgical deletion — no residual references to removed features
- `with parsl_session(config):` without `as session` is idiomatic when the yielded value is unused
- Unconditional `_shutdown_parsl()` in `finally` is safer than the conditional guard
- All callers verified — no `build_systems(wait=False)` calls anywhere in the codebase
- Completeness check: all 4 acceptance criteria from issue 52 are met

**Agents run:** code-reviewer (0 findings), completeness-checker (all met), simplifier (3 findings)

---
*Reviewed by mach6*

**Comment by gregorweiss** [comment]
## Review Assessment

[Review comment](https://github.com/emdgroup/mdfactory/pull/53#issuecomment-5624182350)

### Classifications

| Finding | Classification | Reasoning |
|---|---|---|
| 1: Leftover "Finding N" comments in tests | **Deferred** | **Factual:** Confirmed — 7 occurrences of "Finding N" prefixes in test section headers. **Scope:** Pre-existing code not touched by this PR. The PR scope is removing `wait=False` / `detach()`, not cleaning up test comments. |
| 2: Duplicate `error`/`error_detail` keys | **Deferred** | **Factual:** Confirmed — both keys hold identical `error_detail` string in `_poll_future`. **Scope:** Pre-existing code in `_poll_future`, not introduced or modified by this PR. The duplicate may serve downstream consumers expecting either key. |
| 3: Mid-file imports with noqa suppressions | **Deferred** | **Factual:** Confirmed — two `# noqa: E402` imports at lines 117 and 156. **Scope:** Pre-existing code structure not touched by this PR. |

### Action Plan

No genuine issues. All 3 findings are factually valid observations about pre-existing code that fall outside the PR's authorized scope.

**0 genuine / 0 nitpick / 0 false positive / 3 deferred**

---
*Assessment by mach6*

**Comment by gregorweiss** [comment]
YAGNI removal

**Review by maxscheurer — APPROVED** [review]
